addons/product_expiry/models/production_lot.py

```diff
@@ -34,21 +34,38 @@ def _get_dates(self, product_id=None):
         res = dict.fromkeys(mapped_fields.keys(), False)
         product = self.env['product.product'].browse(product_id) or self.product_id
         if product:
+            op_date = datetime.datetime.now()
+            if life_date:
+                res['life_date'] = fields.Datetime.from_string(life_date)
+                op_date = res['life_date'] - datetime.timedelta(days=product.life_time)
+
             for field in mapped_fields.keys():
                 duration = getattr(product, mapped_fields[field])
                 if duration:
-                    date = datetime.datetime.now() + datetime.timedelta(days=duration)
+                    date = op_date + datetime.timedelta(days=duration)
                     res[field] = fields.Datetime.to_string(date)
         return res
 
     # Assign dates according to products data
     @api.model
     def create(self, vals):
-        vals.update(self._get_dates(vals.get('product_id')))
+        vals.update(self._get_dates(vals.get('product_id'), vals.get('life_date')))
         return super(StockProductionLot, self).create(vals)
 
     @api.onchange('product_id')
     def _onchange_product(self):
         dates_dict = self._get_dates()
         for field, value in dates_dict.items():
             setattr(self, field, value)
+
+
+class PackOperationLot(models.Model):
+    _inherit = "stock.pack.operation.lot"
+
+    def _default_life_date(self):
+        if self.env.context.get('product_id'):
+            life_days = self.env['product.product'].browse(self.env.context['product_id']).life_time
```

Comment:
Actually **product_id** field is not present in current model, so we need to pass it from context.